src/react-native/21-publishing-and-updating/publishing-and-updating.md

```diff
@@ -10,33 +10,83 @@
 
-TODO
+```
+MYAPP_UPLOAD_STORE_FILE=my-upload-key.keystore # Replace with the name of your keystore file
+MYAPP_UPLOAD_KEY_ALIAS=my-key-alias # Replace with the alias you set
+MYAPP_UPLOAD_STORE_PASSWORD=your-keystore-password  # Replace with the keystore password you set
+MYAPP_UPLOAD_KEY_PASSWORD=your-key-password  # Replace with the key password you set
+```
 
-### Concept TODO 2
+Lastly, we will upadte the `android/app/build.gradle` file to add the following:
 
-TODO
+```gradle
+...
+android {
+    ...
+    defaultConfig { ... }
+    signingConfigs {
+        release {
+            if (project.hasProperty('MYAPP_UPLOAD_STORE_FILE')) {
+                storeFile file(MYAPP_UPLOAD_STORE_FILE)
+                storePassword MYAPP_UPLOAD_STORE_PASSWORD
+                keyAlias MYAPP_UPLOAD_KEY_ALIAS
+                keyPassword MYAPP_UPLOAD_KEY_PASSWORD
+            }
+        }
+    }
+    buildTypes {
+        release {
+            ...
+            signingConfig signingConfigs.release
+        }
+    }
+}
+...
+```
 
-## Objective 3: Updating an existing application
+The comprehensive guide to signing an AAB can be found [here](https://reactnative.dev/docs/signed-apk-android).
 
-TODO
+Now that we've generated an upload key and configured Gradle to use it, we can generate an AAB file and upload it to the Google Play store.
 
-### Concept TODO 3
+```bash
+npx react-native build-android --mode="release"
+```
 
-TODO
+## Objective 2: App updates and versioning
+
+### Releasing updates
+
+The two main ways to release updates to your app are to either upload a new AAB to the Google Play store or to create a new OTA (over-the-air) update.
+
+OTA updates allow us to update our app without requiring the user to install a new version from the app store. OTA updates are great for quick deployment for bug fixes or small changes. However, they are limited to JavaScript changes only. The overhead of implementing an OTA process is higher than simply uploading a new AAB to the app store.
+
+Deploying a new AAB to the Google Play store is more suited for updates beyond small bug fixes or changes. With a new AAB we can update native code. Users are familiar with the update process, so deploying a new AAB with a changelog can provide a better user experience when doing major updates.
+
+Both OTA and AAB updates have their place in the app update process. It's up to you to decide whether one or the other, or a mix of both, is best for your app.
+
+### Testing new releases
+
+Both the Google Play store and the Apple App store allow you to test new releases before making them available to the public. This is a great way to ensure that your app is working as expected before releasing it to the public.
```

Comment:
I think “Store” should always be capitalized when used with either the Google Play Store or App Store.

Reply:
fixed